File: 0649-dota2-senate/0649-dota2-senate.py
class Solution:
    def predictPartyVictory(self, senate: str) -> str:
        radiant = collections.deque()
        dire = collections.deque()

        for i in range(len(senate)):
            if senate[i] == "R":
                radiant.append(i)
            else:
                dire.append(i)
        
        while radiant and dire:
            first, second = radiant.popleft(), dire.popleft()

            if first < second:
                radiant.append(first + len(senate))
            else:
                dire.append(second + len(senate))

        return "Radiant" if radiant else "Dire"

File: 0649-dota2-senate/0649-dota2-senate.go
func predictPartyVictory(senate string) string {
	radiants, dires := []int{}, []int{}
	for i, c := range senate {
		if c == 'R' {
			radiants = append(radiants, i)
		} else {
			dires = append(dires, i)
		}
	}

	for len(radiants) > 0 && len(dires) > 0 {
		r, d := radiants[0], dires[0]
		radiants, dires = radiants[1:], dires[1:]
		if r < d {
			radiants = append(radiants, r+len(senate))
		} else {
			dires = append(dires, d+len(senate))
		}
	}

	if len(radiants) > 0 {
		return "Radiant"
	}

	return "Dire"
}